app/pages/project/floating-ips/FloatingIpsPage.tsx

```diff
@@ -0,0 +1,242 @@
+    title="No Floating IPs"
+    body="You need to create a Floating IP to be able to see it here"
+    buttonText="New Floating IP"
+    buttonTo={pb.floatingIpNew(useProjectSelector())}
+  />
+)
+
+FloatingIpsPage.loader = async ({ params }: LoaderFunctionArgs) => {
+  const { project } = getProjectSelector(params)
+  await Promise.all([
+    apiQueryClient.prefetchQuery('floatingIpList', {
+      query: { project, limit: 25 },
+    }),
+    apiQueryClient.prefetchQuery('instanceList', {
+      query: { project },
+    }),
+  ])
+  return null
+}
+
+export function FloatingIpsPage() {
+  const [attachModalOpen, setAttachModalOpen] = useState(false)
+  const [floatingIpToModify, setFloatingIpToModify] = useState<FloatingIp | null>(null)
+  const queryClient = useApiQueryClient()
+  const { project } = useProjectSelector()
+  const { data: instances } = usePrefetchedApiQuery('instanceList', {
+    query: { project },
+  })
+  const getInstanceName = (instanceId: string) =>
+    instances.items.find((i) => i.id === instanceId)?.name
+
+  const floatingIpDetach = useApiMutation('floatingIpDetach', {
+    onSuccess() {
+      queryClient.invalidateQueries('floatingIpList')
+      addToast({ content: 'Your Floating IP has been detached' })
+    },
+    onError: (err) => {
+      addToast({ title: 'Error', content: err.message, variant: 'error' })
+    },
+  })
+  const deleteFloatingIp = useApiMutation('floatingIpDelete', {
+    onSuccess() {
+      queryClient.invalidateQueries('floatingIpList')
+      addToast({ content: 'Your Floating IP has been deleted' })
+    },
+  })
+
+  const makeActions = (floatingIp: FloatingIp): MenuAction[] => {
+    const isAttachedToAnInstance = !!floatingIp.instanceId
+    return [
+      {
+        label: 'Attach',
+        disabled: isAttachedToAnInstance
+          ? 'This floating IP must be detached from the existing instance before it can be attached to a new one'
+          : false,
+        onActivate() {
+          setFloatingIpToModify(floatingIp)
+          setAttachModalOpen(true)
+        },
+      },
+      {
+        label: 'Detach',
+        disabled: isAttachedToAnInstance
+          ? false
+          : 'This floating IP is not attached to an instance',
+        onActivate: () =>
+          confirmAction({
+            actionType: 'danger',
+            doAction: () =>
+              floatingIpDetach.mutateAsync({
+                path: { floatingIp: floatingIp.name },
+                query: { project },
+              }),
+            modalTitle: 'Detach Floating IP',
+            modalContent: (
+              <p>
+                Are you sure you want to detach the floating IP <HL>{floatingIp.name}</HL>{' '}
+                from the instance{' '}
+                <HL>
+                  {
+                    // instanceId is guaranteed to be non-null here
+                    getInstanceName(floatingIp.instanceId!)
+                  }
+                </HL>
+                ?
```

Comment:
I would add another line here like `The instance will no longer be reachable at <HL>{floatingIp.ip}</HL>`." And it's a nitpick, but I think both "the"s are unnecessary. `Are you sure you want to detach floating IP cola-float from instance db1` reads ok to me.

<img width="532" alt="image" src="https://github.com/oxidecomputer/console/assets/3612203/e09fe508-3481-4086-b158-5267e0ab9ae9">